File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        int i = 0;
        int j = s.length()-1;
        char[] c = s.toCharArray();
        while(i < j){
            while( i < j && !isVowel(c[i])){
                i++;
            }
            while(i < j && !isVowel(c[j])){
                j--;
            }

            while (i >= j){
                break;
            }
            swap(c, i, j);
            i++;
            j--;
        }

        return new String(c);
    }
    private boolean isVowel(char c){
        if (c == 'a'|| c =='e' || c == 'i'|| c =='o' ||c == 'u'|| c =='A' ||c == 'E'|| c =='I' ||c == 'O'|| c =='U' ){
       return true;
        }
        return false;
    }
   public void swap(char[] c, int i, int j){
        char temp = c[i];
        c[i] = c[j];
        c[j] = temp;
    }
}

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function (s) {
    let vowels = ['a', 'e', 'i', 'o', 'u'];
    let arr = s.split('');
    let l = 0, r = s.length - 1;

    while (l < r) {
        while (l < r && !vowels.includes(arr[l].toLowerCase())) l++;
        while (l < r && !vowels.includes(arr[r].toLowerCase())) r--;

        // swap vowels
        let temp = arr[l];
        arr[l] = arr[r];
        arr[r] = temp;

        l++;
        r--;
    }

    return arr.join('');
};